pmd-java/src/test/kotlin/net/sourceforge/pmd/lang/java/types/ast/ConditionalTypeTest.kt

```diff
@@ -16,17 +16,15 @@ class ConditionalTypeTest : FunSpec({}) {
 
     private val tested = mutableMapOf<TypePair, JTypeMirror>()
 
-    fun test(t1: JTypeMirror, t2: JTypeMirror, expected: JTypeMirror) {
+    private fun runTest(t1: JTypeMirror, t2: JTypeMirror, expected: JTypeMirror) {
         val key = TypePair(t1, t2)
         if (key in tested && tested[key] != expected)
             throw AssertionError("Already tested $t1 : $t2 against ${tested[key]}, doesn't match $expected")
         else if (key in tested) return
 
         tested[key] = expected
 
-        test("$t1 : $t2 => $expected") {
-            PolyResolution.computeStandaloneConditionalType(testTypeSystem, t1, t2) shouldBe expected
-        }
+        PolyResolution.computeStandaloneConditionalType(testTypeSystem, t1, t2) shouldBe expected
```

Comment:
I wonder why you pulled up the `test` statement? These tests used to be more granular, now if one of these assertions fails we don't know which one. If you want to keep it that way, I think you can replace this `test("...") {...}` with `withClue("...") {...}` so that we have a better error message -but in the previous way we would also have directly all failed/passed assertions in the IDE.

Reply:
That was necessary to have them running :)
It seems, the new kotest version made the generators suspend, so that the test were defined _after_ the test class has been instantiated. And that's not possible anymore - test can't be added after the test class has been fully instantiated (maybe that's the change in kotest, who knows?).
Anyway - the change in kotest seems to be that all the tests need to be known before executing them (which makes a progress bar possible if you don't execute against a moving target...).

I can give it a try with `withClue`...